---
title: "Gene Enrichment Analysis with MSigDB"
author: "Angela Yoder"
date: "`r Sys.Date()`"
output: 
  html_document:
    code_folding: hide
---

**The goals of this program are:**

1.  Identify rat gene sets
2.  Convert Ensembl Gene IDs into Entrez Ids
3.  Report on missingness in Biomart
3.  Use clusterProfiler to do pathway analysis

For our gene set enrichment analysis (GSEA), we will be using the data resource MSigDB (Molecular Signatures Database), "a collection of annotated gene sets for use with GSEA software." This dataset was used in conjunction with the GSEA R package clusterProfiler. 
We identified our enrichment dataset via the function msigdbr from the msigdbr r package. The species was identified as "Rattus norvegicus." msigdbr has access to multiple collections, including hallmark gene sets (H), and eight different supplementary gene sets (C1 to C8). In this analysis, we focused on the Hallmark Sets (H), KEGG Pathways (part of C2), and GO: Biological Processes (part of C5). 

```{r}
library(msigdbr)

H_genes = msigdbr(species = "Rattus norvegicus", category = "H")
C2_genes = msigdbr(species = "Rattus norvegicus", category = "C2") # HAS KEGG genes
C5_genes = msigdbr(species = "Rattus norvegicus", category = "C5") #GO genes
#C8_genes = msigdbr(species = "Rattus norvegicus", category = "C8")

# https://cran.r-project.org/web/packages/msigdbr/vignettes/msigdbr-intro.html
# https://bioconductor.org/packages/release/bioc/manuals/clusterProfiler/man/clusterProfiler.pdf

```

```{r}

#setwd("C:/Users/angel/OneDrive/Desktop/Saba Lab/Data/")
setwd("C:/Users/yodeange/OneDrive - The University of Colorado Denver/Saba Lab/Data")

tesec_brain = read.csv("Target_TESec_Ssec_brain.csv")
tesec_liver = read.csv("Target_TESec_Ssec_liver.csv")
tensec_brain = read.csv("Target_TENSec_Ssec_brain.csv")
tensec_liver = read.csv("Target_TENSec_Ssec_liver.csv")
bicor_brain = read.csv("Bicor_brain.csv")
bicor_liver = read.csv("Bicor_liver.csv")

fix.read.csv = function(x)
{
  names = x[,1]
  gee = x[,2:dim(x)[2]]
  rownames(gee) = names
  return(gee)
}
`%!in%` = Negate(`%in%`)

tesec_brain = fix.read.csv(tesec_brain)
tesec_liver = fix.read.csv(tesec_liver)
tensec_brain = fix.read.csv(tensec_brain)
tensec_liver = fix.read.csv(tensec_liver)
bicor_brain = fix.read.csv(bicor_brain)
bicor_liver = fix.read.csv(bicor_liver)

# top Ssec genes for brain and liver FROM: SeldinReport_fix.Rmd

top.brain = tesec_brain[1:6,]
top.liver = tesec_liver[1,]

```
The top Tissue-Exclusive, Secreted (TES) protein-coding genes were determined in SeldinReport_fix.Rmd by determining which TES protein-coding genes had Ssec scores higher than the 95th percentile of Ssec scores from tissue-exclusive, non-secreted (TENS) protein-coding genes. We then identified the target tissue transcripts that were most significantly associated with the top TES protein-coding genes (p-value < 0.05). These will be the gene sets for which we will identify annotations.  

```{r}

# genes they are correlated to < 0.05

top.corr.brain = bicor_brain[rownames(bicor_brain) %in% rownames(top.brain),]
top.corr.liver = bicor_liver[rownames(bicor_liver) %in% rownames(top.liver),]

list.oflists = function(mat, num)
{
  out = list(c(), c(), c(), c(), c(), c())
  out = out[1:num]
  if(num > 1)
  {
    for (i in 1:num)
    {
      save = colnames(mat[,mat[i,] < 0.05])
      out[[i]] = save
    }
  }
  else
  {
    save = colnames(mat[,mat[1,] < 0.05])
    out[[1]] = save
  }
  return(out)
}

brain.list = list.oflists(top.corr.brain, nrow(top.brain))
liver.list = list.oflists(top.corr.liver, nrow(top.liver))

combine.brain = unique(c(brain.list[[1]], brain.list[[2]], brain.list[[3]], brain.list[[4]], brain.list[[5]], brain.list[[6]]))

combine.liver = liver.list[[1]]

num.brain = c(length(brain.list[[1]]), length(brain.list[[2]]), length(brain.list[[3]]), length(brain.list[[4]]), length(brain.list[[5]]), length(brain.list[[6]]))

num.liver = c(length(liver.list[[1]]))

num.target.brain = data.frame(top.brain[,1], num.brain)
num.target.liver = data.frame(top.liver[,1], num.liver)
colnames(num.target.brain) = c("TES PC from Brain", "Number of Target Genes in Liver")
colnames(num.target.liver) = c("TES PC from Liver", "Number of Target Genes in Brain")

#knitr::kable(num.target.brain, caption = "Number of Target Genes in the Liver associated with TES Protein-Coding Genes from the Brain")

#knitr::kable(num.target.liver, caption = "Number of Target Genes in the Brain associated with TES Protein-Coding Genes from the Liver")

setwd("C:/Users/yodeange/OneDrive - The University of Colorado Denver/Saba Lab/Data")

summ.brain = read.csv("Summary_Table_topSsec_brain.csv")
summ.liver = read.csv("Summary_Table_topSsec_liver.csv")

summ.brain = fix.read.csv(summ.brain)
summ.liver = fix.read.csv(summ.liver)

knitr::kable(summ.brain, caption = "Summary of Top Ssec Scores for Brain")
knitr::kable(summ.liver, caption = "Summary of Top Ssec Score for Liver")

```

Entrez gene IDs were identified via biomart version 104. 

## Missingness Report

When retrieving Entrez IDs for our gene sets of interest, I found that many of the genes within these sets were no longer available within biomaRt. Looking at the Ensembl.org website, many of these ensembl IDs were retired via Ensembl. The version in which these ensembl IDs were available is version 104. This version was used for the enrichment analysis. However, there may be concerns using these retired ensembl IDs, especially where there is no longer a corresponding ID in the new version. Below is a table summarizing the number of genes missing, along with examples of pages from defunct ensembl ID webpages. 

```{r, echo = FALSE, eval = FALSE}

library(biomaRt)

#ensembl = useMart(biomart = "ENSEMBL_MART_ENSEMBL", dataset = "rnorvegicus_gene_ensembl", version = 109)
ensembl = useEnsembl(biomart="ensembl",dataset = "rnorvegicus_gene_ensembl")
ensembl = useEnsembl(biomart="ensembl",dataset = "rnorvegicus_gene_ensembl", version = 104)

get_attrib <- listAttributes(ensembl)
#go_attrib <- get_attrib[grep("GO ",get_attrib$description),]

# getting entrez ids
get_entrez = getBM(attributes=c("ensembl_gene_id",
                              "external_gene_name",
                              "description",
                              "chromosome_name",
                              "start_position",
                              "end_position",
                              "gene_biotype",
                            #"entrezgene_trans_name",
                            "entrezgene_id"),
                 filters = 'biotype',
                 values = "protein_coding",
                 mart = ensembl)


entrez_brain.1= get_entrez[get_entrez$ensembl_gene_id %in% combine.brain,]

entrez_topbrain = get_entrez[get_entrez$ensembl_gene_id %in% top.brain$Gene_symbol,]
entrez_topliver = get_entrez[get_entrez$ensembl_gene_id %in% top.liver$Gene_symbol,]

#sum(get_uniprot$ensembl_gene_id %in% rownames(ts_Ssec_target_brain))
#GO_fin = subset(get_uniprot, get_uniprot$ensembl_gene_id %in% rownames(ts_Ssec_target_brain))

entrez_liver.1 = get_entrez[get_entrez$ensembl_gene_id %in% combine.liver,]

# missing gene ids

missing_b = combine.brain[combine.brain %!in% entrez_brain.1$ensembl_gene_id]
# 15 missing TES : "ENSRNOG00000047739" "ENSRNOG00000008563" "ENSRNOG00000009140" "ENSRNOG00000038437" "ENSRNOG00000034130" "ENSRNOG00000003887" "ENSRNOG00000007374" "ENSRNOG00000013441" "ENSRNOG00000003891" "ENSRNOG00000038980" "ENSRNOG00000013228" "ENSRNOG00000038150" "ENSRNOG00000049171" "ENSRNOG00000011074" "ENSRNOG00000050172"

# 66 missing target :  ENSRNOG00000031343, ENSRNOG00000037928, ENSRNOG00000046284, ENSRNOG00000046386, ENSRNOG00000047103, ENSRNOG00000052003, ENSRNOG00000056194, ENSRNOG00000057089, ENSRNOG00000058753, ENSRNOG00000058824, ENSRNOG00000020628, ENSRNOG00000030575, ENSRNOG00000037495, ENSRNOG00000042197, ENSRNOG00000043176, ENSRNOG00000045991, ENSRNOG00000046572, ENSRNOG00000047308, ENSRNOG00000047560, ENSRNOG00000048017, ENSRNOG00000049009, ENSRNOG00000049299, ENSRNOG00000049315, ENSRNOG00000050243, ENSRNOG00000053899, ENSRNOG00000055193, ENSRNOG00000055375, ENSRNOG00000055620, ENSRNOG00000058460, ENSRNOG00000059007, ENSRNOG00000059199, ENSRNOG00000060451, ENSRNOG00000013464, ENSRNOG00000033245, ENSRNOG00000046696, ENSRNOG00000047079, ENSRNOG00000047535, ENSRNOG00000048070, ENSRNOG00000048182, ENSRNOG00000050066, ENSRNOG00000050118, ENSRNOG00000050981, ENSRNOG00000056285, ENSRNOG00000057734, ENSRNOG00000058352, ENSRNOG00000060328, ENSRNOG00000003648, ENSRNOG00000042083, ENSRNOG00000057548, ENSRNOG00000061431, ENSRNOG00000033130, ENSRNOG00000042543, ENSRNOG00000046529, ENSRNOG00000049727, ENSRNOG00000050499, ENSRNOG00000003498, ENSRNOG00000039931, ENSRNOG00000040195, ENSRNOG00000047843, ENSRNOG00000050232, ENSRNOG00000050885, ENSRNOG00000051227, ENSRNOG00000054524, ENSRNOG00000058034, ENSRNOG00000013410, ENSRNOG00000046011

missing_l = combine.liver[combine.liver %!in% entrez_liver.1$ensembl_gene_id]
# 12 missing TES : "ENSRNOG00000020805" "ENSRNOG00000050499" "ENSRNOG00000049727" "ENSRNOG00000033130" "ENSRNOG00000042543" "ENSRNOG00000046529" "ENSRNOG00000013464" "ENSRNOG00000049515" "ENSRNOG00000013410" "ENSRNOG00000016535" "ENSRNOG00000033619" "ENSRNOG00000033245"

# 34 missing target : "ENSRNOG00000007374" "ENSRNOG00000047739" "ENSRNOG00000000060" "ENSRNOG00000006051" "ENSRNOG00000008125" "ENSRNOG00000011369" "ENSRNOG00000012057" "ENSRNOG00000012086" "ENSRNOG00000012961" "ENSRNOG00000024149" "ENSRNOG00000025745" "ENSRNOG00000028114" "ENSRNOG00000029046" "ENSRNOG00000032206" "ENSRNOG00000038184" "ENSRNOG00000039197" "ENSRNOG00000040213" "ENSRNOG00000042602" "ENSRNOG00000042609" "ENSRNOG00000042788" "ENSRNOG00000042939" "ENSRNOG00000043120" "ENSRNOG00000047075" "ENSRNOG00000047740" "ENSRNOG00000048172" "ENSRNOG00000048358" "ENSRNOG00000049878" "ENSRNOG00000050511" "ENSRNOG00000051471" "ENSRNOG00000055889" "ENSRNOG00000056130" "ENSRNOG00000057009" "ENSRNOG00000057522" "ENSRNOG00000061301"


#setwd("C:/Users/angel/OneDrive/Desktop/Saba Lab/Data/")
setwd("C:/Users/yodeange/OneDrive - The University of Colorado Denver/Saba Lab/Data")
write.csv(entrez_topbrain, "EntrezIds_topbrain.csv")
write.csv(entrez_topliver, "EntrezIds_topliver.csv")
write.csv(entrez_brain.1, "EntrezIds_brain.csv")
write.csv(entrez_liver.1, "EntrezIds_liver.csv")

```

```{r, echo = FALSE}

missingness = data.frame(dataset = c("TES Brain", "Sig Target Genes Brain", "TES Liver", "Sig Target Genes Liver"), missing = c(15, 66, 12, 34), total = c(174, 182, 35, 329))

knitr::kable(missingness, caption = "Missing Ensembl IDs within Datasets")

#missing.tes.liver = c("ENSRNOG00000020805", "ENSRNOG00000050499", "ENSRNOG00000049727", "ENSRNOG00000033130", "ENSRNOG00000042543", "ENSRNOG00000046529", "ENSRNOG00000013464", "ENSRNOG00000049515", "ENSRNOG00000013410", "ENSRNOG00000016535", "ENSRNOG00000033619", "ENSRNOG00000033245")

```

![Missing Gene: ENSRNOG00000050885](C:/Users/yodeange/OneDrive - The University of Colorado Denver/Saba Lab/Data/ENSRNOG00000050885_Ensembl.png)

![Missing Gene: ENSRNOG00000013464](C:/Users/yodeange/OneDrive - The University of Colorado Denver/Saba Lab/Data/ENSRNOG00000013464_Ensembl.png)


```{r, echo = FALSE, eval = FALSE}

knitr::kable(missing.tes.liver, caption = "Missing TES Liver Ensembl IDs") 

#setwd("C:/Users/angel/OneDrive/Desktop/Saba Lab/Data/")
setwd("C:/Users/yodeange/OneDrive - The University of Colorado Denver/Saba Lab/Data")

miss.tes.liver = read.csv("Missing_Ensembl_TES_Liver.txt")

knitr::kable(miss.tes.liver, caption = "Ensembl ID History Converter")

```

## Remapping Missing Ensembl IDs

Due to this missingness, efforts were made to identify new Ensembl IDs for these missing genes. Updated Ensembl IDs for the missing IDs were retrieved via Ensembl.org's ID History Converter, whose output was subsetted by updates to Version 105. A decent portion of our missing Ensembl IDs had corresponding IDs in the new version, however, many were divided into multiple IDs. These were not able to be converted one-to-one due to uncertainty about which version of the new IDs our detected genes would map to. This resulted in only one target gene from the brain and one target gene from the liver having a viable one-to-one match to a Version 105 Ensembl ID. 

```{r}

## we got this from http://uswest.ensembl.org/Rattus_norvegicus/Tools/IDMapper?db=core;expand_form=true;tl=veVyFBLEW339Hw8k-8445349

setwd("C:/Users/yodeange/OneDrive - The University of Colorado Denver/Saba Lab/Data")

remap.tesbrain = read.csv("remapping_TESbrain.csv")
remap.tesliver = read.csv("remapping_TESliver.csv")
remap.targetbrain = read.csv("remapping_targetbrain.csv")
remap.targetliver = read.csv("remapping_targetliver.csv")

remap.tesbrain$Releases = sub("\\:.*", "", remap.tesbrain$Releases)
remap.targetbrain$Releases = sub("\\:.*", "", remap.targetbrain$Releases)
remap.tesliver$Releases = sub("\\:.*", "", remap.tesliver$Releases)
remap.targetliver$Releases = sub("\\:.*", "", remap.targetliver$Releases)

remap.tesbrain.105 = remap.tesbrain[remap.tesbrain$Releases == 105, ]
remap.targetbrain.105 = remap.targetbrain[remap.targetbrain$Releases == 105, ]
remap.tesliver.105 = remap.tesliver[remap.tesliver$Releases == 105, ]
remap.targetliver.105 = remap.targetliver[remap.targetliver$Releases == 105, ]

```

```{r}

# summary

summ.remap = data.frame(dataset = missingness$dataset, total.genes = missingness$total, missing.genes = missingness$missing, num.remapped = c(nrow(remap.tesbrain.105), nrow(remap.targetbrain.105), nrow(remap.tesliver.105), nrow(remap.targetliver.105)), num.unique.remapped = c(length(unique(remap.tesbrain.105$Requested.ID)), length(unique(remap.targetbrain.105$Requested.ID)), length(unique(remap.tesliver.105$Requested.ID)), length(unique(remap.targetliver.105$Requested.ID))))

#knitr::kable(summ.remap)

```

```{r}

# usable remapping

only.one = function(x)
{
  out = c()
  for (i in 1:length(x))
  {
    if (x[i] %in% x[-i])
    {
      out = out
    }
    else
    {
      out = c(out, x[i])
    }
  }
  return(out)
}

usable.tesbrain = remap.tesbrain.105[remap.tesbrain.105$Requested.ID %in% only.one(remap.tesbrain.105$Requested.ID),]
usable.targetbrain = remap.targetbrain.105[remap.targetbrain.105$Requested.ID %in% only.one(remap.targetbrain.105$Requested.ID),]
usable.tesliver = remap.tesliver.105[remap.tesliver.105$Requested.ID %in% only.one(remap.tesliver.105$Requested.ID), ]
usable.targetliver = remap.targetliver.105[remap.targetliver.105$Requested.ID %in% only.one(remap.targetliver.105$Requested.ID),]

one.to.one.match = c(length(usable.tesbrain$Requested.ID), length(usable.targetbrain$Requested.ID), length(usable.tesliver$Requested.ID), length(usable.targetliver$Requested.ID))

summ.remap = data.frame(summ.remap, one.to.one.match)

knitr::kable(summ.remap)

```

```{r, echo = FALSE, eval = FALSE}

# getting entrez IDs for these Ensembl IDs
# replace ensembl in combine brain and then rerun

replace.id = function(final, replace)
{
  for(i in 1:length(replace$Requested.ID))
  {
    final[final == replace$Requested.ID[i]] = replace$Matched.ID.s[i]
  }
  return(final)
}

combine.brain.remap = replace.id(combine.brain, usable.targetbrain)
combine.liver.remap = replace.id(combine.liver, usable.targetliver)

# run biomart to get entrez IDS

library(biomaRt)

ensembl = useEnsembl(biomart="ensembl",dataset = "rnorvegicus_gene_ensembl")

get_entrez = getBM(attributes=c("ensembl_gene_id",
                              "external_gene_name",
                              "description",
                              "chromosome_name",
                              "start_position",
                              "end_position",
                              "gene_biotype",
                            #"entrezgene_trans_name",
                            "entrezgene_id"),
                 filters = 'biotype',
                 values = "protein_coding",
                 mart = ensembl)


entrez_brain.1= get_entrez[get_entrez$ensembl_gene_id %in% combine.brain,]

entrez_liver.1 = get_entrez[get_entrez$ensembl_gene_id %in% combine.liver,]

# compare to other dataframe

setwd("C:/Users/yodeange/OneDrive - The University of Colorado Denver/Saba Lab/Data")
entrez_brain = read.csv("EntrezIds_brain.csv")
entrez_liver = read.csv("EntrezIds_liver.csv")

entrez_brain = fix.read.csv(entrez_brain)
entrez_liver = fix.read.csv(entrez_liver)


```

## Enrichment Analysis

Summary of enrichment analysis was complicated by the R function enricher not including fold enrichment in its output. I have included a summary table of the number of enriched terms for each top TES protein-coding gene in this analysis. At the end of the document, there are also plots provided in the r package enrichplot, though they would only correspond to a single TES protein-coding gene's analysis within a single gene set. I am unsure how useful they would be in describing these relationships. 

```{r, message = FALSE, warning = FALSE}

#setwd("C:/Users/angel/OneDrive/Desktop/Saba Lab/Data/")
setwd("C:/Users/yodeange/OneDrive - The University of Colorado Denver/Saba Lab/Data")
entrez_brain = read.csv("EntrezIds_brain.csv")
entrez_liver = read.csv("EntrezIds_liver.csv")

entrez_brain = fix.read.csv(entrez_brain)
entrez_liver = fix.read.csv(entrez_liver)

#msigdbr_t2g = msigdbr_df %>% dplyr::distinct(gs_name, entrez_gene) %>% as.data.frame()
#enricher(gene = gene_ids_vector, TERM2GENE = msigdbr_t2g, ...)

msigdbr_t2g_H = H_genes %>% dplyr::distinct(gs_name, entrez_gene) %>% as.data.frame()
msigdbr_t2g_C2 = C2_genes %>% dplyr::distinct(gs_name, entrez_gene) %>% as.data.frame()
msigdbr_t2g_C5 = C5_genes %>% dplyr::distinct(gs_name, entrez_gene) %>% as.data.frame()
#msigdbr_t2g_C8 = C8_genes %>% dplyr::distinct(gs_name, entrez_gene) %>% as.data.frame()


library(clusterProfiler)

# pAdjustMethod, min/max (number?) of genes

#enricher(gene = target_genes, TERM2GENE = msigdbr_t2g, ...)

# H

out.brain_H = list(c(), c(), c(), c(), c(), c())
database = msigdbr_t2g_H

for (i in 1:6)
{
  vect = brain.list[[i]]
  entrez = entrez_brain$entrezgene_id[entrez_brain$ensembl_gene_id %in% vect]
  entrez = na.omit(entrez)
  listt = enricher(gene = entrez, TERM2GENE = database, pAdjustMethod = "fdr")
  out.brain_H[[i]] = data.frame(ID = listt$ID, pvalue = listt$pvalue, p.adjust = listt$p.adjust, geneID = listt$geneID)
}

out.liver_H = list(c())


  vect = liver.list[[1]]
  entrez = entrez_liver$entrezgene_id[entrez_liver$ensembl_gene_id %in% vect]
  entrez = na.omit(entrez)
  listt = enricher(gene = entrez, TERM2GENE = database, pAdjustMethod = "fdr")
  out.liver_H[[1]] = data.frame(ID = listt$ID, pvalue = listt$pvalue, p.adjust = listt$p.adjust, geneID = listt$geneID)



#brain.ids = c(paste(out.brain[[1]]$ID, collapse = ", "), paste(out.brain[[2]]$ID, collapse = ", "), paste(out.brain[[3]]$ID, collapse = ", "), paste(out.brain[[4]]$ID, collapse = ", "), paste(out.brain[[5]]$ID, collapse = ", "))
#names(brain.ids) = top.brain[,1]

#knitr::kable(brain.ids, caption = "Enrichment of Target Liver Genes Influenced by TES Protein-Coding Genes from the Brain", colnames = "")

#liver.ids = c(paste(out.liver[[1]]$ID, collapse = ", "), paste(out.liver[[2]]$ID, collapse = ", "), paste(out.liver[[3]]$ID, collapse = ", "), paste(out.liver[[4]]$ID, collapse = ", "), paste(out.liver[[5]]$ID, collapse = ", "))
#names(liver.ids) = top.liver[,1]

#knitr::kable(liver.ids, caption = "Enrichment of Target Liver Genes Influenced by TES Protein-Coding Genes from the Liver", colnames = "")

# C2

out.brain_C2 = list(c(), c(), c(), c(), c(), c())
database = msigdbr_t2g_C2

for (i in 1:6)
{
  vect = brain.list[[i]]
  entrez = entrez_brain$entrezgene_id[entrez_brain$ensembl_gene_id %in% vect]
  entrez = na.omit(entrez)
  listt = enricher(gene = entrez, TERM2GENE = database, pAdjustMethod = "fdr")
  out.brain_C2[[i]] = data.frame(ID = listt$ID, pvalue = listt$pvalue, p.adjust = listt$p.adjust, geneID = listt$geneID)
}

out.liver_C2 = list(c())


  vect = liver.list[[1]]
  entrez = entrez_liver$entrezgene_id[entrez_liver$ensembl_gene_id %in% vect]
  entrez = na.omit(entrez)
  listt = enricher(gene = entrez, TERM2GENE = database, pAdjustMethod = "fdr")
  out.liver_C2[[1]] = data.frame(ID = listt$ID, pvalue = listt$pvalue, p.adjust = listt$p.adjust, geneID = listt$geneID)



#brain.ids = c(paste(out.brain[[1]]$ID, collapse = ", "), paste(out.brain[[2]]$ID, collapse = ", "), paste(out.brain[[3]]$ID, collapse = ", "), paste(out.brain[[4]]$ID, collapse = ", "), paste(out.brain[[5]]$ID, collapse = ", "))
#names(brain.ids) = top.brain[,1]

#knitr::kable(brain.ids, caption = "Enrichment of Target Liver Genes Influenced by TES Protein-Coding Genes from the Brain", colnames = "")

#liver.ids = c(paste(out.liver[[1]]$ID, collapse = ", "), paste(out.liver[[2]]$ID, collapse = ", "), paste(out.liver[[3]]$ID, collapse = ", "), paste(out.liver[[4]]$ID, collapse = ", "), paste(out.liver[[5]]$ID, collapse = ", "))
#names(liver.ids) = top.liver[,1]

#knitr::kable(liver.ids, caption = "Enrichment of Target Liver Genes Influenced by TES Protein-Coding Genes from the Liver", colnames = "")

# C5

out.brain_C5 = list(c(), c(), c(), c(), c(), c())
database = msigdbr_t2g_C5

for (i in 1:6)
{
  vect = brain.list[[i]]
  entrez = entrez_brain$entrezgene_id[entrez_brain$ensembl_gene_id %in% vect]
  entrez = na.omit(entrez)
  listt = enricher(gene = entrez, TERM2GENE = database, pAdjustMethod = "fdr")
  out.brain_C5[[i]] = data.frame(ID = listt$ID, pvalue = listt$pvalue, p.adjust = listt$p.adjust, geneID = listt$geneID)
}

out.liver_C5 = list(c())


  vect = liver.list[[1]]
  entrez = entrez_liver$entrezgene_id[entrez_liver$ensembl_gene_id %in% vect]
  entrez = na.omit(entrez)
  listt = enricher(gene = entrez, TERM2GENE = database, pAdjustMethod = "fdr")
  out.liver_C5[[1]] = data.frame(ID = listt$ID, pvalue = listt$pvalue, p.adjust = listt$p.adjust, geneID = listt$geneID)

```

```{r}
# looking for correlations to TES also

setwd("C:/Users/yodeange/OneDrive - The University of Colorado Denver/Saba Lab/Data")
entrez_topbrain = read.csv("EntrezIds_topbrain.csv")
entrez_topliver = read.csv("EntrezIds_topliver.csv")

database = msigdbr_t2g_H
two.H = enricher(gene = c(entrez_topbrain$entrezgene_id[2], 315323, 171154), TERM2GENE = database, pAdjustMethod = "fdr")
#two.H$geneID

five.H1 = enricher(gene = c(c(entrez_topbrain$entrezgene_id[5], 50549, 84029, 25598, 29171)), TERM2GENE = database, pAdjustMethod = "fdr")
#five.H1$geneID

five.H2 = enricher(gene = c(c(entrez_topbrain$entrezgene_id[5], 266708, 29187, 309178, 171371)), TERM2GENE = database, pAdjustMethod = "fdr")
#five.H2$geneID

database = msigdbr_t2g_C2
five.c2 = enricher(gene = c(c(entrez_topbrain$entrezgene_id[5], 50549, 25598, 29171)), TERM2GENE = database, pAdjustMethod = "fdr")
#five.c2$geneID

six.c2 = enricher(gene = c(c(entrez_topbrain$entrezgene_id[6], 291283, 307092, 289533, 63867)), TERM2GENE = database, pAdjustMethod = "fdr")
#six.c2$geneID

database = msigdbr_t2g_C5
one.c5 = enricher(gene = c(c(entrez_topbrain$entrezgene_id[1], 296366, 312102, 301229, 289057)), TERM2GENE = database, pAdjustMethod = "fdr")
#one.c5$geneID

two.c5 = enricher(gene = c(c(entrez_topbrain$entrezgene_id[2], 29300, 117033, 25098)), TERM2GENE = database, pAdjustMethod = "fdr")
#two.c5$geneID
# "311862/117033"

three.c5 = enricher(gene = c(c(entrez_topbrain$entrezgene_id[3], 24833, 408236, 81806, 171154)), TERM2GENE = database, pAdjustMethod = "fdr")
#three.c5$geneID

six.c5 = enricher(gene = c(c(entrez_topbrain$entrezgene_id[6], 291283, 311387, 307092)), TERM2GENE = database, pAdjustMethod = "fdr")
#six.c5$geneID

database = msigdbr_t2g_H
liver.H = enricher(gene = c(c(entrez_topliver$entrezgene_id, 24294, 689619, 79208, 29471, 24898, 25412, 25546, 25041, 29361)), TERM2GENE = database, pAdjustMethod = "fdr")
#liver.H$geneID

################# LOOK INTO THIS ONE #####################
database = msigdbr_t2g_C2
liver.c2 = enricher(gene = c(c(entrez_topliver$entrezgene_id, 65187, 24409, 289606, 25153, 24182, 24410, 140675, 25033, 29595, 116506, 24316, 81721, 24922, 29471, 25674, 24418, 29234, 170796, 25041)), TERM2GENE = database, pAdjustMethod = "fdr")
#liver.c2$geneID

database = msigdbr_t2g_C5
liver.c5 = enricher(gene = c(c(entrez_topliver$entrezgene_id, 65187, 24409, 689043, 289606, 114032, 60337, 50599, 170899, 24410, 140675, 117018, 307206, 24316, 81721, 24922, 29161, 294388, 29701, 25674, 170739, 24602, 78956, 24418, 307540, 25041)), TERM2GENE = database, pAdjustMethod = "fdr")
#liver.c5$geneID

```

### Enrichment of Target Liver Protein-Coding Genes acted upon by TES Protein-Coding Genes from Brain

```{r}

hallmark.brain = c(nrow(out.brain_H[[1]]), nrow(out.brain_H[[2]]), nrow(out.brain_H[[3]]), nrow(out.brain_H[[4]]), nrow(out.brain_H[[5]]), nrow(out.brain_H[[6]]))

these_1k = grep("KEGG", out.brain_C2[[1]][,1], value = TRUE)
these_2k = grep("KEGG", out.brain_C2[[2]][,1], value = TRUE)
these_3k = grep("KEGG", out.brain_C2[[3]][,1], value = TRUE)
these_4k = grep("KEGG", out.brain_C2[[4]][,1], value = TRUE)
these_5k = grep("KEGG", out.brain_C2[[5]][,1], value = TRUE)
these_6k = grep("KEGG", out.brain_C2[[6]][,1], value = TRUE)

kegg.brain = c(nrow(out.brain_C2[[1]][out.brain_C2[[1]][,1] %in% these_1k,]), 
               nrow(out.brain_C2[[2]][out.brain_C2[[2]][,1] %in% these_2k,]), 
               nrow(out.brain_C2[[3]][out.brain_C2[[3]][,1] %in% these_3k,]), 
               nrow(out.brain_C2[[4]][out.brain_C2[[4]][,1] %in% these_4k,]), 
               nrow(out.brain_C2[[5]][out.brain_C2[[5]][,1] %in% these_5k,]), 
               nrow(out.brain_C2[[6]][out.brain_C2[[6]][,1] %in% these_6k,]))

these_1g = grep("GOBP", out.brain_C5[[1]][,1], value = TRUE)
these_2g = grep("GOBP", out.brain_C5[[2]][,1], value = TRUE)
these_3g = grep("GOBP", out.brain_C5[[3]][,1], value = TRUE)
these_4g = grep("GOBP", out.brain_C5[[4]][,1], value = TRUE)
these_5g = grep("GOBP", out.brain_C5[[5]][,1], value = TRUE)
these_6g = grep("GOBP", out.brain_C5[[6]][,1], value = TRUE)


go.brain = c(nrow(out.brain_C5[[1]][out.brain_C5[[1]][,1] %in% these_1g,]), 
               nrow(out.brain_C5[[2]][out.brain_C5[[2]][,1] %in% these_2g,]), 
               nrow(out.brain_C5[[3]][out.brain_C5[[3]][,1] %in% these_3g,]), 
               nrow(out.brain_C5[[4]][out.brain_C5[[4]][,1] %in% these_4g,]), 
               nrow(out.brain_C5[[5]][out.brain_C5[[5]][,1] %in% these_5g,]), 
               nrow(out.brain_C5[[6]][out.brain_C5[[6]][,1] %in% these_6g,]))

summary.brain = data.frame(gene.symbol = summ.brain$Gene_Symbol, hallmark.terms = hallmark.brain, kegg.terms = kegg.brain, go.bio.process.terms = go.brain, sig.target.genes = summ.brain$Number_Target_Genes)

knitr::kable(summary.brain, caption = "Number of Significant Terms for Each Top Brain TES Protein-Coding Genes")

```

#### HALLMARK GENE SET

```{r}

#knitr::kable(out.brain_H[[1]], caption = "Hallmark Brain TES Protein-Coding Gene: ENSRNOG00000005652")
knitr::kable(out.brain_H[[2]], caption = "2. Hallmark Brain TES Protein-Coding Gene: ENSRNOG00000002548")
#knitr::kable(out.brain_H[[3]], caption = "Hallmark Brain TES Protein-Coding Gene: ENSRNOG00000002381")
#knitr::kable(out.brain_H[[4]], caption = "Hallmark Brain TES Protein-Coding Gene: ENSRNOG00000003845")
knitr::kable(out.brain_H[[5]], caption = "5. Hallmark Brain TES Protein-Coding Gene: ENSRNOG00000059507")
#knitr::kable(out.brain_H[[6]], caption = "Hallmark Brain TES Protein-Coding Gene: ENSRNOG00000000010")

```

#### C2 GENE SET

```{r}
# Search KEGG

#knitr::kable(out.brain_C2[[1]][out.brain_C2[[1]][,1] %in% these_1k,], caption = "C2: KEGG Brain TES Protein-Coding Gene: ENSRNOG00000005652")
#knitr::kable(out.brain_C2[[2]][out.brain_C2[[2]][,1] %in% these_2k,], caption = "C2: KEGG Brain TES Protein-Coding Gene: ENSRNOG00000002548")
#knitr::kable(out.brain_C2[[3]][out.brain_C2[[3]][,1] %in% these_3k,], caption = "C2: KEGG Brain TES Protein-Coding Gene: ENSRNOG00000002381")
#knitr::kable(out.brain_C2[[4]][out.brain_C2[[4]][,1] %in% these_4k,], caption = "C2: KEGG Brain TES Protein-Coding Gene: ENSRNOG00000003845")
knitr::kable(out.brain_C2[[5]][out.brain_C2[[5]][,1] %in% these_5k,], caption = "5. C2: KEGG Brain TES Protein-Coding Gene: ENSRNOG00000059507")
knitr::kable(out.brain_C2[[6]][out.brain_C2[[6]][,1] %in% these_6k,], caption = "6. C2: KEGG Brain TES Protein-Coding Gene: ENSRNOG00000000010")

```

#### C5 GENE SET

```{r}

knitr::kable(out.brain_C5[[1]][out.brain_C5[[1]][,1] %in% these_1g,], caption = "1. C5: GO Biological Process Brain TES Protein-Coding Gene: ENSRNOG00000005652")
knitr::kable(out.brain_C5[[2]][out.brain_C5[[2]][,1] %in% these_2g,], caption = "2. C5: GO Biological Process Brain TES Protein-Coding Gene: ENSRNOG00000002548")
knitr::kable(out.brain_C5[[3]][out.brain_C5[[3]][,1] %in% these_3g,], caption = "3. C5: GO Biological Process Brain TES Protein-Coding Gene: ENSRNOG00000002381")
#knitr::kable(out.brain_C5[[4]][out.brain_C5[[4]][,1] %in% these_4g,], caption = "C5: GO Biological Process Brain TES Protein-Coding Gene: ENSRNOG00000003845")
#knitr::kable(out.brain_C5[[5]][out.brain_C5[[5]][,1] %in% these_5g,], caption = "C5: GO Biological Process Brain TES Protein-Coding Gene: ENSRNOG00000059507")
knitr::kable(out.brain_C5[[6]][out.brain_C5[[6]][,1] %in% these_6g,], caption = "6. C5: GO Biological Process Brain TES Protein-Coding Gene: ENSRNOG00000000010")

```

### Enrichment of Target Brain Protein-Coding Genes acted upon by TES Protein-Coding Genes from Liver

```{r}

hallmark.liver = nrow(out.liver_H[[1]])

these_1k = grep("KEGG", out.liver_C2[[1]][,1], value = TRUE)
kegg.liver = nrow(out.liver_C2[[1]][out.liver_C2[[1]][,1] %in% these_1k,])

these_1g = grep("GOBP", out.liver_C5[[1]][,1], value = TRUE)
go.liver = nrow(out.liver_C5[[1]][out.liver_C5[[1]][,1] %in% these_1g,])

summary.liver = data.frame(gene.symbol = summ.liver$Gene_Symbol, hallmark.terms = hallmark.liver, kegg.terms = kegg.liver, go.bio.process.terms = go.liver, sig.target.genes = summ.liver$Number_Target_Genes)

knitr::kable(summary.liver, caption = "Number of Significant Terms for Each Top Liver TES Protein-Coding Genes")

```

#### HALLMARK GENE SET

```{r}

knitr::kable(out.liver_H[[1]], caption = "Hallmark Liver TES Protein-Coding Gene: ENSRNOG00000047481")

```

#### C2 GENE SET

```{r}
# search KEGG

knitr::kable(out.liver_C2[[1]][out.liver_C2[[1]][,1] %in% these_1k,], caption = "C2: KEGG Liver TES Protein-Coding Gene: ENSRNOG00000047481")

```

#### C5 GENE SET

```{r}

knitr::kable(out.liver_C5[[1]][out.liver_C5[[1]][,1] %in% these_1g,], caption = "C5: GO Biological Processes Liver TES Protein-Coding Gene: ENSRNOG00000047481")

```

### Other Plots

```{r}

library(enrichplot)
barplot(listt)
cnetplot(listt) 
dotplot(listt)
#goplot(listt) #DAG

```

### Cursory Googling

TES.brain.2.Hallmark: Hallmark_KRAS_SIGNALING_DN
  https://www.gsea-msigdb.org/gsea/msigdb/cards/HALLMARK_KRAS_SIGNALING_DN.html
  Genes down-regulated by KRAS activation
  K-RAS protein: part of signalling pathway that give instructions for the cell to grow and divide or mature and take on specialized functions
  KRAS + alcohol linked to pancreatic cancer
  
TES.brain.5.Hallmark: HALLMARK_FATTY_ACID_METABOLISM
  https://www.gsea-msigdb.org/gsea/msigdb/cards/HALLMARK_FATTY_ACID_METABOLISM
  Genes encoding proteins involved in metabolism of fatty acids
  https://pubmed.ncbi.nlm.nih.gov/87483/#:~:text=Progressive%20alteration%20of%20the%20mitochondria,production%2C%20which%20results%20in%20ketonemia. "Progressive alteration of the mitochondria, which occurs during chronic alcohol consumption, decreases fatty acid oxidation by interfering with citric acid cycle activity. This block is partially compensated for by increased ketone body production, which results in ketonemia."

TES.brain.5.Hallmark: HALLMARK_INTERFERON_GAMMA_RESPONSE
  https://www.gsea-msigdb.org/gsea/msigdb/cards/HALLMARK_INTERFERON_GAMMA_RESPONSE.html
  Genes up-regulated in response to IFNG
  INFG protein: secreted as a part of innate and adaptive immune systems
  Cytokines and lipid peroxidation in alcoholics with chronic hepatitis C virus infection: https://academic.oup.com/alcalc/article/43/2/137/122851

TES.brain.5.C2-KEGG: KEGG_PPAR_SIGNALING_PATHWAY
  https://www.gsea-msigdb.org/gsea/msigdb/cards/KEGG_PPAR_SIGNALING_PATHWAY
  Peroxisome proliferator-activated receptors- nuclear hormone receptors activated by fatty acids
  liver pathway: https://www.genome.jp/kegg/pathway/hsa/hsa03320.html
  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4308472/ Peroxisome proliferator-activated receptors alpha and gamma are linked with alcohol consumption in mice and withdrawal and dependence in humans

TES.liver.1.Hallmark: HALLMARK_KRAS_SIGNALLING_DN
  * see TES.brain.2.Hallmark
  
TES.liver.1.C2-KEGG: KEGG_NEUROACTIVE_LIGAND_RECEPTOR_INTERACTION
  pathway: https://www.genome.jp/kegg/pathway/hsa/hsa04080.html
  
  



