---
title: "Report for Secreted Proteins"
author: "Angela Yoder"
date: "`r Sys.Date()`"
---

**The goals of this program are:**

1. Report results from Data Cleaning.Rmd and Annotation.Rmd
Total secreted in rat genome
Introduction on secreted proteins. Where got them from. How found them in my data. 
Could make a series of reports of the two Rmd
Overlap between the two tissues

```{r, echo = FALSE, message = FALSE}
# Packages
library(tidyverse)

# Data Input
setwd("C:/Users/angel/OneDrive/Desktop/Saba Lab/Data/")
protein_brain = read.csv("Brain_WGCNA.csv")
secreted_protein_brain = read.csv("Secreted_Brain.csv")
protein_liver = read.csv("Liver_WGCNA.csv")
secreted_protein_liver = read.csv("Secreted_Liver.csv")

fin.r.squared_brain = read.csv("R_Squared_Brain.csv")
fin.r.squared_brain_sec = read.csv("R_Squared_Brain_Sec.csv")

funct_brain = read.csv("Function_Annotation_Brain.csv")
funct_brain_sec = read.csv("Function_Annotation_Brain_Sec.csv")


fix.read.csv = function(x)
{
  names = x[,1]
  gee = x[,2:dim(x)[2]]
  rownames(gee) = names
  return(gee)
}

protein_brain = fix.read.csv(protein_brain)
secreted_protein_brain = fix.read.csv(secreted_protein_brain)
fin.r.squared_brain = fix.read.csv(fin.r.squared_brain)
fin.r.squared_brain_sec = fix.read.csv(fin.r.squared_brain_sec)
funct_brain = fix.read.csv(funct_brain)
funct_brain_sec = fix.read.csv(funct_brain_sec)
protein_liver = fix.read.csv(protein_liver)
secreted_protein_liver = fix.read.csv(secreted_protein_liver)

```

```{r, echo = FALSE, message = FALSE}
# table comparing number of genes
ngenes = ncol(protein_brain)
ngenes_sec = ncol(secreted_protein_brain)

ngenes.l = ncol(protein_liver)
ngenes_sec.l = ncol(secreted_protein_liver)

num = c(ngenes, ngenes_sec, ngenes.l, ngenes_sec.l)

names(num) = c("Brain: Full Data", "Brain: Secreted Data", "Liver: Full Data", "Liver: Secreted Data")

knitr::kable(num, caption = "Number of Genes", col.names = NA, )

```

Density plots and overlaying. Compare medians, quartiles- boxplot- ttest - might not show biological relavence. - use confidence intervals nonparametric test
```{r, echo = FALSE, message = FALSE}
# Hist R Squared

hist(as.numeric(fin.r.squared_brain_sec[,2]), main = "Broad Sense Heritability: Secreted Brain", xlab = "R-Squared Value")
hist(as.numeric(fin.r.squared_brain[,2]), main = "Broad Sense Heritability: Full Brain", xlab = "R-Squared Value")

```

exclude cellular components - no good- enrichment more useful later on
eqtl : find point in geneome changing expression; usually near it- transcription factor binding site; P30 webinars- intro to expression QTL and QTL; go to every snp; ttest of quant value at each snp; looking for really low pvalue. rpackage called qtl- need to get data in right format- tutorials- email; genotype dataset in email (full_hrdp_geno)- BIGGER project
```{r, echo = FALSE, message = FALSE}
# Annotation Comparison

most = slice_max(funct_brain_sec, order_by = funct_brain_sec$Freq, n = 30)
murst = slice_max(funct_brain, order_by = funct_brain$Freq, n = 30)
most = most[1:30,]

most = most[order(most[,3], decreasing = TRUE),]
murst = murst[order(murst[,3], decreasing = TRUE),]

fin = cbind(as.character(most[,1]), round(most[,3], digits = 3), as.character(murst[,1]), round(murst[,3], digits = 3))
colnames(fin) = c("Full: Protein Function", "Full: Proportion", "Secreted: Protein Function", "Secreted: Proportion")

knitr::kable(fin)

```