---
title: "Report for Secreted Proteins"
author: "Angela Yoder"
date: "`r Sys.Date()`"
---

**The goals of this program are:**

1. Report results of secreted proteins data

The original data of ensembl rat liver and brain RNAseq data was downloaded from BioMart database (accessed 2021). Inbred strains were excluded, leaving only recombinant inbred strains. Enrichment data was retrieved from BiomaRt (version number `r packageVersion("biomaRT")`), which was used to exclude non-protein-coding genes from the data set. This resulted in a dataset of 30 strains and 16,457 genes for brain tissue, and 30 strains and 14,109 genes for liver tissue. 

From there, the data was further reduced to only secreted proteins. This was done using a list of secreted proteins from the uniprot database (accessed 2021). Results for protein-coding genes were filtered by species "Rattus norvegicus" and subcellular location "secreted." This resulted in a list of 1,772 secreted proteins attributed to Rattus norvegicus. The uniprot IDs from the new secreted dataset were compared to the converted uniprot IDs (via biomaRt) of the brain and liver tissue datasets. This comparison in 657 secreted proteins from the brain dataset, and 518 secreted proteins from the liver dataset.

The following report was completed using `r R.Version()$version.string`. Additionally, the packages WGCNA (version `r packageVersion("WGCNA")`), broom (version `r packageVersion("broom")`), tidyverse (version `r packageVersion("tidyverse")`), and knitr (version `r packageVersion("knitr")`). 


```{r, echo = FALSE, message = FALSE}
# Packages
library(tidyverse)

# Data Input
setwd("C:/Users/angel/OneDrive/Desktop/Saba Lab/Data/")
protein_brain = read.csv("Brain_WGCNA.csv")
secreted_protein_brain = read.csv("Secreted_Brain.csv")
protein_liver = read.csv("Liver_WGCNA.csv")
secreted_protein_liver = read.csv("Secreted_Liver.csv")
#notsec_protein_brain = read.csv("NotSecreted_Brain.csv")
#notsec_protein_liver = read.csv("NotSecreted_Liver.csv")

fin.r.squared_brain = read.csv("R_Squared_Brain.csv")
fin.r.squared_brain_sec = read.csv("R_Squared_Brain_Sec.csv")
fin.r.squared_liver = read.csv("R_Squared_Liver.csv")
fin.r.squared_liver_sec = read.csv("R_Squared_Liver_Sec.csv")
fin.r.squared_brain_notsec = read.csv("R_Squared_Brain_NotSec.csv")
fin.r.squared_liver_notsec = read.csv("R_Squared_Liver_NotSec.csv")

#funct_brain = read.csv("Function_Annotation_Brain.csv")
#funct_brain_sec = read.csv("Function_Annotation_Brain_Sec.csv")


fix.read.csv = function(x)
{
  names = x[,1]
  gee = x[,2:dim(x)[2]]
  rownames(gee) = names
  return(gee)
}

protein_brain = fix.read.csv(protein_brain)
secreted_protein_brain = fix.read.csv(secreted_protein_brain)
fin.r.squared_brain = fix.read.csv(fin.r.squared_brain)
fin.r.squared_brain_sec = fix.read.csv(fin.r.squared_brain_sec)
#funct_brain = fix.read.csv(funct_brain)
#funct_brain_sec = fix.read.csv(funct_brain_sec)
protein_liver = fix.read.csv(protein_liver)
secreted_protein_liver = fix.read.csv(secreted_protein_liver)
#notsec_protein_brain = fix.read.csv(notsec_protein_brain)
#notsec_protein_liver = fix.read.csv(notsec_protein_liver)
fin.r.squared_liver = fix.read.csv(fin.r.squared_liver)
fin.r.squared_liver_sec = fix.read.csv(fin.r.squared_liver_sec)
fin.r.squared_brain_notsec = fix.read.csv(fin.r.squared_brain_notsec)
fin.r.squared_liver_notsec = fix.read.csv(fin.r.squared_liver_notsec)

```

The following tables show the number of proteins for each dataset. Here we can see that secreted proteins make up a small proportion of genes expressed in the liver and brain tissues. Of the brain proteins, secreted proteins make up approximately 4%; of the liver proteins, secreted proteins make up 3.8%. Additionally, of all secreted proteins found in the rat proteome, those produced by the brain and liver are 37% and 29% respectively. There is also a significant overlap in secreted proteins in the brain and liver. In the second table, we can see that 74% of proteins secreted by the brain are also secreted by the liver, and 93% vice versa. 

```{r, echo = FALSE, message = FALSE}
# table comparing number of genes
ngenes = ncol(protein_brain)
ngenes_sec = ncol(secreted_protein_brain)

ngenes.l = ncol(protein_liver)
ngenes_sec.l = ncol(secreted_protein_liver)
tot_sec = 1760 # from secreted uniprot data

num = c(ngenes, ngenes_sec, ngenes.l, ngenes_sec.l, tot_sec)

names(num) = c("Brain: Full Data", "Brain: Secreted Data", "Liver: Full Data", "Liver: Secreted Data", "Total Secreted Rat Proteins")

knitr::kable(num, caption = "Number of Genes", col.names = NA, )

# compare shared genes between secreted brain and secreted liver

gee = intersect(colnames(secreted_protein_brain), colnames(secreted_protein_liver))

int = c(length(colnames(secreted_protein_brain)), length(gee), length(colnames(secreted_protein_liver)))
names(int) = c("Secreted Proteins Brain", "Secreted Proteins Shared by Brain and Liver", "Secreted Proteins Liver")

knitr::kable(int)

```

```{r, echo = FALSE, message = FALSE}
# hypothesis test wilcoxon mann whitney

#brain

gee = cbind(rep("not secreted", times = dim(fin.r.squared_brain_notsec)[1]), fin.r.squared_brain_notsec[,2])
gee2 = cbind(rep("secreted", times = dim(fin.r.squared_brain_sec)[1]), fin.r.squared_brain_sec[,2])

wil.b = rbind(gee, gee2)
wil.b = as.data.frame(wil.b)
colnames(wil.b) = c("trt", "r.sq")

wil.b$trt = as.factor(wil.b$trt)
wil.b$r.sq = as.numeric(wil.b$r.sq)

wil.brain = coin::wilcox_test(r.sq ~ trt, wil.b)

# liver

gee = cbind(rep("not secreted", times = dim(fin.r.squared_liver_notsec)[1]), fin.r.squared_liver_notsec[,2])
gee2 = cbind(rep("secreted", times = dim(fin.r.squared_liver_sec)[1]), fin.r.squared_liver_sec[,2])

wil.l = rbind(gee, gee2)
wil.l = as.data.frame(wil.l)
colnames(wil.l) = c("trt", "r.sq")

wil.l$trt = as.factor(wil.l$trt)
wil.l$r.sq = as.numeric(wil.l$r.sq)

wil.liver = coin::wilcox_test(r.sq ~ trt, wil.l)

```

The following boxplots and density plots show an increase in median value and a decrease in right skew that may indicate that the secreted dataset has higher relative heritability.
 
```{r, echo = FALSE, message = FALSE}
# Hist R Squared

#hist(as.numeric(fin.r.squared_brain_sec[,2]), main = "Broad Sense Heritability: Secreted Brain", xlab = "R-Squared Value")
#hist(as.numeric(fin.r.squared_brain[,2]), main = "Broad Sense Heritability: Full Brain", xlab = "R-Squared Value")

#hist(as.numeric(fin.r.squared_liver_sec[,2]), main = "Broad Sense Heritability: Secreted Liver", xlab = "R-Squared Value")
#hist(as.numeric(fin.r.squared_liver[,2]), main = "Broad Sense Heritability: Full Liver", xlab = "R-Squared Value")

# Density R Squared

ggplot(wil.b, aes(x = r.sq, color = trt)) + 
  geom_density() +
  labs(title="R Squared Density Curve: Brain",x="R Squared", y = "Density")

ggplot(wil.l, aes(x = r.sq, color = trt)) + 
  geom_density() +
  labs(title="R Squared Density Curve: Liver",x="R Squared", y = "Density")

# Box R Squared

boxplot(fin.r.squared_brain[,2], fin.r.squared_brain_sec[,2], names = c("Not Secreted", "Secreted"), xlab = "R Squared", main = "R Squared Brain Data")
boxplot(fin.r.squared_liver[,2], fin.r.squared_liver_sec[,2], names = c("Not Secreted", "Secreted"), xlab = "R Squared", main = "R Squared Liver Data")

```

We tested whether the heritability of the non-secreted and secreted datasets were significantly different using the Wilcoxon-Mann-Whitney Test. Both the tests of the brain and liver data showed that the two datasets were significantly different at an alpha = 0.05 level. 

```{r, echo = FALSE, message = FALSE}
print("Non-Secreted Brain Dataset v. Secreted Brain Dataset")
wil.brain

print("Non-Secreted Liver Dataset v. Secreted Liver Dataset")
wil.liver

R.Version()$version.string

```


